src/NzbDrone.Common/OAuth/WebParameterCollection.cs

```diff
@@ -14,14 +14,14 @@ public class WebParameterCollection : IList<WebParameter>
         {
             get
             {
-                var parameters = this.Where(p => p.Name.Equals(name));
+                var parameters = this.Where(p => p.Name.Equals(name)).ToArray();
```

Comment:
Here we actually filtered by name three times by calling Count() twice and by ToArray call